### PR TITLE
Backgroundcolor

### PR DESCRIPTION
**Pull request by beattie**
Add support to goatplot to set the background color used by the draw function to draw the background.

### CODE REVIEW

File: src/goat-plot.c [hunk 1]
```diff
@@ -610,7 +624,32 @@ goat_plot_set_ticks_y (GoatPlot *plot, gdouble major, gint minors_per_major)
...
+void
+goat_plot_set_background_color(GoatPlot *plot, GdkRGBA *color)
+{
+	g_return_if_fail (plot);
+	g_return_if_fail (GOAT_IS_PLOT (plot));
+
+	GoatPlotPrivate *priv;
+
+	priv = GOAT_PLOT_GET_PRIVATE (plot);
+
```
Comment (drahnr) [inline src/goat-plot.c]: handling of 'color == NULL'?

**Comment by beattie** [comment]
A thought occurs to me, would setting color to NULL be usefull.  Also threr are two ways to handle a NULL. Set to a default, or ignore.  For now I'll fix this so trying to set the color with a NULL as an argument will be ignored.  Will look into returning an error and/or issuing a Warning

**Comment by drahnr** [comment]
NULL could be indeed reset to default, but this is just an afterthough. Usually you do not want to handle this in the library but rather in the application, since the application might want to provide a sane default itself.

From my perspective, NULL is definitely an invalid color.

**Comment by drahnr** [comment]
I merged this manually.

**Pull request closed without merging**